src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static Figure getRandomFigure() {
+        Random random = new Random();
+        int index = random.nextInt(RandomFigure.values().length);
+        RandomFigure randomFigure = RandomFigure.values()[index];
+        Figure figure;
+        switch (randomFigure) {
+            case CIRCLE:
+                figure = new Circle();
```

Comment:
you should use constructor with parameters here, and pass all values through it.
`new Circle(randomRadius, randomColor);`

*to get `randomRadius` use `random`, to get `randomColor` use `colorSupplier.getRandomColor()`